From e0106af25d938cd194a82beca923e0b4607885aa Mon Sep 17 00:00:00 2001
Message-Id: <e0106af25d938cd194a82beca923e0b4607885aa.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Akashdeep Sharma <akashdeep.sharma@intel.com>
Date: Mon, 2 Feb 2015 10:07:28 +0530
Subject: [PATCH 03/61] MUST_REBASE [VPG]: video/adf: Register ADF with LPE
 Audio
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

If ADF is enabled, register ADF driver with HDMI LPE audio driver
instead of i915.

Also, moved/added function definitions from i915_adf.h to 
i915_adf_wrapper.c so as to avoid build errors in hdmi_audio_if.c 
if CONFIG_ADF_INTEL is not set.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5532
Change-Id: I0da81533b64f1af1bd61b4d38dc06c91376d73ec
Signed-off-by: Akashdeep Sharma <akashdeep.sharma@intel.com>
---
 drivers/gpu/drm/i915/Makefile           |    5 ++---
 drivers/gpu/drm/i915/hdmi_audio_if.c    |    8 ++++++++
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    6 ++++++
 include/drm/i915_adf.h                  |   10 ++++------
 4 files changed, 20 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/i915/Makefile b/drivers/gpu/drm/i915/Makefile
index 2be6314..034a849 100644
--- a/drivers/gpu/drm/i915/Makefile
+++ b/drivers/gpu/drm/i915/Makefile
@@ -76,14 +76,13 @@ i915-y += dvo_ch7017.o \
 	  intel_tv.o \
 	  i915_dpst.o \
 	  intel_clrmgr.o \
-	  i915_sprite_csc.o
+	  i915_sprite_csc.o \
+	  i915_adf_wrapper.o
 
 # legacy horrors
 i915-y += i915_dma.o \
 	  i915_ums.o
 
-i915-$(CONFIG_ADF_INTEL) += i915_adf_wrapper.o
-
 obj-$(CONFIG_DRM_I915)  += i915.o
 
 obj-$(CONFIG_DRM_I915_SYNC)  += intel_sync.o
diff --git a/drivers/gpu/drm/i915/hdmi_audio_if.c b/drivers/gpu/drm/i915/hdmi_audio_if.c
index f153441..2226996 100644
--- a/drivers/gpu/drm/i915/hdmi_audio_if.c
+++ b/drivers/gpu/drm/i915/hdmi_audio_if.c
@@ -26,6 +26,7 @@
  */
 
 #include <drm/drmP.h>
+#include <i915_adf.h>
 #include "hdmi_audio_if.h"
 #include "i915_drv.h"
 #include "i915_reg.h"
@@ -339,6 +340,10 @@ int mid_hdmi_audio_setup(
 	struct drm_i915_private *dev_priv;
 	int ret = 0;
 
+	if (i915.enable_intel_adf && g_adf_ready)
+		return adf_hdmi_audio_setup((void *)audio_callbacks,
+				(void *)reg_ops, (void *)query_ops);
+
 	DRM_DEBUG_DRIVER("%s: called\n", __func__);
 
 	if (!hdmi_priv)
@@ -372,6 +377,9 @@ int mid_hdmi_audio_register(struct snd_intel_had_interface *driver,
 	struct drm_device *dev;
 	struct drm_i915_private *dev_priv;
 
+	if (i915.enable_intel_adf && g_adf_ready)
+		return adf_hdmi_audio_register((void *)driver, had_data);
+
 	DRM_DEBUG_DRIVER("%s: called\n", __func__);
 	if (!hdmi_priv)
 		return -ENODEV;
diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 1a6597f..2c6a233 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -223,4 +223,10 @@ unsigned short *intel_get_vbt_pps_delays(void)
 	return (unsigned short *)&i915_adf_dev->vbt.edp_pps;
 }
 EXPORT_SYMBOL(intel_get_vbt_pps_delays);
+#else
+int intel_adf_context_on_event(void) { return 0; }
+
+/* ADF register calls for audio driver */
+int adf_hdmi_audio_register(void *drv, void *had_data) { return 0; }
+int adf_hdmi_audio_setup(void *callbacks, void *r_ops, void *q_ops) { return 0; }
 #endif
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 9beb1ad..e0806ce 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -36,12 +36,6 @@
 #define INTEL_SIDEBAND_REG_READ		0
 #define INTEL_SIDEBAND_REG_WRITE	1
 
-#ifndef CONFIG_ADF_INTEL
-volatile bool g_adf_ready = false;
-
-static int intel_adf_context_on_event(void) { return 0; }
-#endif
-
 extern void intel_adf_dpio_sideband_rw(u32 operation, u32 port,
 				       u32 reg, u32 *val);
 extern void intel_adf_pci_sideband_rw(u32 operation, u32 port,
@@ -61,6 +55,10 @@ extern u16 intel_get_dsi_port_frm_vbt(void);
 extern void intel_get_vbt_disp_conf(void **child_dev, int *child_dev_num);
 extern struct i2c_adapter *intel_adf_get_gmbus_adapter(u8 port);
 extern unsigned short *intel_get_vbt_pps_delays(void);
+
+/* ADF register calls for audio driver */
+int adf_hdmi_audio_register(void *drv, void *had_data);
+int adf_hdmi_audio_setup(void *callbacks, void *r_ops, void *q_ops);
 #endif
 
 
-- 
1.7.9.5

